test/e2e/nvidia-container-toolkit_test.go

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package e2e
+
+import (
+	"context"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+// Integration tests for Docker runtime
+var _ = Describe("docker", func() {
+	// GPUs are accessible in a container: Running nvidia-smi -L inside the
+	// container shows the same output inside the container as outside the
+	// container. This means that the following commands must all produce
+	// the same output
+	When("running nvidia-smi -L", Ordered, func() {
+		var rawOut string
+
+		BeforeAll(func(ctx context.Context) {
+			_, err := runScript("docker pull ubuntu")
+			Expect(err).ToNot(HaveOccurred())
+
+			rawOut, err = runScript("nvidia-smi -L")
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		Describe("comparing outputs of nvidia-smi -L inside and outside a container", func() {
```

Comment:
Do we need the `Describe` spec? What about:
```
When("running nvidia-smi -L")
   It("should support NVIDIA_VISIBLE_DEVICES")
   It("should support automatic CDI spec generation")
   It("should support the --gpus flag using the nvidia-container-runtime-hook")
   It("should support the --gpus flag using the nvidia-container-runtime")
```

Reply:
Done

Reply:
Why do we want to match the strings to the doc? They should signify the intent, and if that's not clear from the doc then the doc should be updated.

To sumarize: We want these tests to validate that one is able to run nvidia-smi in a container and get the expected result regardless of the mechanism used to inject the devices into the container.

Reply:
ack,  will edit

Reply:
should I also re-write the vectorAdd and deviceQuery tests?

Reply:
ready for review